### ISSUE TITLE
Feature request: Validated parameters provided to nested stack against nested template.

### ISSUE THREAD

**Issue by hauboldj** [aws-cloudformation/cfn-python-lint#1963]
*cfn-lint version: (`cfn-lint --version`)*
0.48.0

*Description of issue.*
I frequently use nested stacks with SAM build. When I lint the root template and the nested stack is setup using `TemplateURL` pointing to a local file, I would like to have cfn-lint check that any parameters in the nested template without a default value are set by parameters in the root stack. Also I would like it to flag any parameters set by the root template that are not present in the nested template.

Checking the parameter types would be an amazing feature but might be difficult, so just matching the parameter names would be sufficient.

**Comment by farski** [comment]
We tend to end up with 3 or 4 layers of nested stacks, and passing a value from a lower layer back to the top and then back down to a different stack happens pretty often. It's also a common reason for stack update failures, since you have to get outputs, template parameters, and stack parameters all updated. It's easy, but also easy to miss something. I have long wanted to request this feature, but didn't have any good ideas on how the linter would know where to look, since the structure of the resources in the stack and the files on disk likely don't match.

If it required some amount of hinting in a YAML comment:

```yaml
# LocalTemplateURL: ./asg.yml
TemplateURL: !Sub ${TemplateUrlPrefix}/asg.yml
```

That'd be work I'd be very happy to do to help catch these kinds of issues.
